rustls/src/client/hs.rs

```diff
@@ -289,16 +305,45 @@ fn emit_client_hello_for_retry(
     // We don't do renegotiation at all, in fact.
     cipher_suites.push(CipherSuite::TLS_EMPTY_RENEGOTIATION_INFO_SCSV);
 
+    let mut chp_payload = ClientHelloPayload {
+        client_version: ProtocolVersion::TLSv1_2,
+        random: input.random,
+        session_id: input.session_id,
+        cipher_suites,
+        compression_methods: vec![Compression::Null],
+        extensions: exts,
+    };
+
+    match (ech_context.as_mut(), config.grease_ech_hpke_provider) {
+        // ECH config, GREASE is irrelevant.
+        (Some(ech_context), _) => {
+            let ech_chp_payload = ech_context.ech_hello(chp_payload, retryreq)?;
```

Comment:
Nit: inline the `ech_chp_payload` binding?

Reply:
Done.